File: two pointers/move_zeroes.py
# two pointers
# tc O(n), sc O(1)
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0
        for r in range(1,len(nums)):
            if nums[l] == 0 and nums[r] != 0:
                #swap
                nums[l], nums[r] = nums[r], nums[l]
            if nums[l] != 0:
                l += 1

File: two pointers/move_zeroes.java
public class move_zeroes {
    public void moveZeroes(int[] nums) {
        int l = 0;
        for(int r = 1; r<nums.length;r++){
            if(nums[l] == 0 && nums[r] != 0){
                int temp = nums[l];
                nums[l] = nums[r];
                nums[r] = temp;
            }
            if(nums[l] != 0){
                l++;
            }
        }
    }
}